examples/nlp/question-answering.py

```diff
@@ -0,0 +1,422 @@
+document, and it must extract the span of text in the document that contains the answer.
+In this case, the model is not computing a probability distribution over answers, but two
+probability distributions over the tokens in the document text, representing the start
+and end of the span containing the answer.
+
+Question answering of this kind is a very challenging NLP task, and the dataset size
+required to train such a model from scratch when the questions and answers are natural
+language is prohibitively huge. As a result, question answering (like almost all NLP
+tasks) benefits enormously from starting from a strong pre-trained foundation model -
+starting from a strong pre-trained language model can reduce the dataset size required to
+reach a given accuracy by multiple orders of magnitude, enabling you to reach very strong
+performance with surprisingly reasonable datasets.
+
+Starting with a pre-trained model adds difficulties, though - where do you get the model
+from? How do you ensure that your input data is preprocessed and tokenized the same way
+as the original model? How do you modify the model to add an output head that matches
+your task of interest?
+
+In this example, we'll show you how to load a model from the Hugging Face [🤗
+Transformers](https://github.com/huggingface/transformers) library to tackle this
+challenge. We'll also load a benchmark question answering dataset from the [🤗
+Datasets](https://github.com/huggingface/datasets) library - this is another open-source
+repository containing a wide range of datasets across many modalities, from NLP to vision
+and beyond. Note, though, that there is no requirement that these libraries must be used
+with each other. If you want to train a model from [🤗
+Transformers](https://github.com/huggingface/transformers) on your own data, or you want
+to load data from [🤗 Datasets](https://github.com/huggingface/datasets) and train your
+own entirely unrelated models with it, that is of course possible (and highly
+encouraged!)
+"""
+
+"""
+## Loading the dataset
+"""
+
+"""
+We will use the [🤗 Datasets](https://github.com/huggingface/datasets) library to download
+the SQUAD question answering dataset using `load_dataset`.
+"""
+
+from datasets import load_dataset
+
+datasets = load_dataset("squad")
+
+"""
+The `datasets` object itself is a
+[`DatasetDict`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#d
+atasetdict), which contains one key for the training, validation and test set. We can see
+the training, validation and test sets all have a column for the context, the question
+and the answers to those questions. To access an actual element, you need to select a
+split first, then give an index. We can see the answers are indicated by their start
+position in the text and their full text, which is a substring of the context as we
+mentioned above. Let's take a look at what a single training example looks like.
+"""
+
+datasets["train"][0]
```

Comment:
The script should also work as a standalone Python script, so use `print()` to display things